packages/core/src/import-models.ts

```diff
@@ -35,6 +35,9 @@ export async function importModels(globPaths: string | string[], modelMatch?: Mo
 
 async function importModelNoGlob(url: string, modelMatch?: ModelMatch): Promise<ModelStatic[]> {
   const module = await import(url);
+  if (module.default && !isModelStatic(module.default) && typeof module.default === 'object') {
+    Object.assign(module, module.default);
```

Comment:
modules are typically sealed, so you need to create a copy like below:

![image](https://github.com/sequelize/sequelize/assets/1280915/6d44a317-a6e6-4df1-b835-d30c4b5ae991)

```suggestion
    module = { ...module, ...module.default };
```